server/controllers/list.js

```diff
@@ -1124,43 +1120,59 @@ const updateListItem = (req, res) => {
         } else {
           editedItemActivity = ActivityType.ITEM_EDIT_DESCRIPTION;
         }
+
+        const data = { listId, userId, itemId, description };
+
+        return returnPayload(updateItem(socketInstance)(data))(doc);
       }
 
       if (isOrdered !== undefined) {
+        itemToUpdate.editedBy = userId;
         itemToUpdate.isOrdered = isOrdered;
         editedItemActivity = isOrdered
           ? ActivityType.ITEM_DONE
           : ActivityType.ITEM_UNHANDLED;
 
-        // TODO: THIS METHOD NEED TO RECOGNIZE CLIENT SOMEHOW,
-        // OTHERWISE ITEMS GETS TOGGLED TWICE BY AUTHOR
-        // const data = { listId, itemId };
-        // changeItemOrderState(socketInstance, dashboardClients, cohortClients)(
-        //   data
-        // );
-      }
+        const data = { listId, userId, itemId, isOrdered };
 
-      if (authorId) {
-        itemToUpdate.authorId = authorId;
+        return returnPayload(
+          updateItem(socketInstance, dashboardClients, cohortClients)(data)
+        )(doc);
       }
 
       if (name) {
         prevItemName = itemToUpdate.name;
         itemToUpdate.name = name;
         editedItemActivity = ActivityType.ITEM_EDIT_NAME;
+        itemToUpdate.editedBy = userId;
+
+        const data = { listId, userId, itemId, name };
+
+        return returnPayload(
+          updateItem(socketInstance, dashboardClients, cohortClients)(data)
+        )(doc);
       }
 
       if (isArchived !== undefined) {
         itemToUpdate.isArchived = isArchived;
-        editedItemActivity = isArchived
-          ? ActivityType.ITEM_ARCHIVE
-          : ActivityType.ITEM_RESTORE;
-      }
+        itemToUpdate.editedBy = userId;
```

Comment:
Why not just assign it unconditionally?

Reply:
Because inside every if I have `return` statement at the end,  so if I put this piece of code outside of all if`s it will never be executed.

Reply:
You could assign it before the first condition is ever run.